MarcusBarnes/mik
# Work on #393.

**Github issue**: (#393)

# What does this Pull Request do?

Adds a utility script that generates sample input for MIK. Useful for contributors who need to supply sample configuration and data as part of pull requests, etc., or possibly even end users who want a set of configuration and data files to base their production files on.

Note that currently only sample data consistent with single-file content models (specifically, .jpg files) is generated. Code to generate sample data for compound, book, and newspapers content models need to be completed. This PR is intended to get consensus around the approach `generate.php` takes. If the approach is satisfactory, we can complete the work necessary to generate output for the other content models.

# What's new?

This PR adds the following files:

```
extras/scripts/samplecontentgenerator/
├── generate.php
├── ini.twig
├── mappings.csv
└── metadata.twig
```

In addition to the `generate.php` script, a sample mappings file and two templates (one for the .ini file and one for the metadata CSV file) allow users to customize the sample input to their needs.

# How should this be tested?

There are no PHPUnit tests for this new tool, so it requires a smoke test:

1. Check out the issue-393 branch.
1. In the MIK directory, run the following command: `php  extras/scripts/samplecontentgenerator/generate.php -id issue393 /tmp/miksampletest`
1. In `/tmp/miksampletest` you should see the following:

```
├── file1.jpg
├── file2.jpg
├── file3.jpg
├── file4.jpg
├── file5.jpg
├── issue393.ini
├── issue393_mappings.csv
└── issue393_metadata.csv
```

The .ini file should contain values that can be used with the sample files in the `/tmp/miksampletest` directory.

Run `php  extras/scripts/samplecontentgenerator/generate.php --help` to see additional command-line options (e.g., using a custom .ini or metadata template).

# Additional Notes

Note that the sample "JPG" files are really text files. MIK doesn't care whether the files are valid JPEG or other binary files. If we want to generate real JPGs, we can borrow the approach the [Islandora Sample Content Generator](https://github.com/mjordan/islandora_scg/blob/7.x/includes/islandora_scg_basic_image.inc#L30) takes to generate valid JPG and other image files.

# Interested parties

@MarcusBarnes, @jpeak5, @bondjimbond and any other current contributors
